src/java.base/share/classes/sun/util/calendar/ZoneInfoFile.java

```diff
@@ -468,10 +468,9 @@ private static ZoneInfo getZoneInfo(String zoneId,
             }
             if (i < savingsInstantTransitions.length) {
                 // javazic writes the last GMT offset into index 0!
-                if (i < savingsInstantTransitions.length) {
```

Comment:
Even my IDE flags it as always true. While it surely is redundant, I wonder if some other check was intended instead. @jodastephen?

Reply:
I'm not sure this is my code - I certainly don't remember it. This PR causes no effective change but be slightly neater, so I guess I approve the PR